x-pack/legacy/plugins/ml/public/application/jobs/jobs_list/components/start_datafeed_modal/time_range_selector/time_range_selector.js

```diff
@@ -216,6 +214,48 @@ function TabStack({ title, items, switchState, switchFunc }) {
     </div>
   );
 }
+
+const DatePickerWithInput = ({ date, onChange, minDate, setIsValid }) => {
+  const [dateString, setDateString] = useState(date.format(TIME_FORMAT));
+
+  function onTextChange(e) {
+    const val = e.target.value;
+    setDateString(val);
+
+    const newDate = moment(val);
+    if (newDate.isValid()) {
+      setIsValid(true);
+      onChange(newDate);
+    } else {
+      setIsValid(false);
+    }
+  }
+
+  function onCalendarChange(newDate) {
+    setDateString(newDate.format(TIME_FORMAT));
+    setIsValid(true);
+    onChange(newDate);
+  }
+
+  return (
+    <>
+      <EuiFieldText
+        value={dateString}
+        onChange={onTextChange}
+        placeholder={TIME_FORMAT}
+        aria-label="CHANGE ME"
```

Comment:
This needs a better label!

Reply:
surely this entices the user to change the date? no?